File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.go
func findMin(nums []int) int {
	left := 0
	right := len(nums) - 1
	result := nums[0]

	for left <= right {
		mid := left + ( right - left ) / 2
		result = min(result, nums[mid])
		if nums[left] <= nums[mid] && nums[left] > nums[right] {
			left = mid + 1
		}else {
			right = mid - 1
		}
	}

	return result
}

File: Data Structures & Algorithms/sort-colors/submission-1.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int left = 0, right = nums.size() - 1;
        for ( int i = 0; i <= right; i++ ) {
            if ( nums[i] == 0 ){
                swap(nums[i], nums[left++]);
            }else if ( nums[i] == 2 ){
                swap(nums[right--], nums[i]);
                i--;
            }
        }
    }
};